app/controllers/api/service_dialogs_controller.rb

```diff
@@ -62,26 +63,26 @@ def set_additional_attributes
       @additional_attributes = %w(content) if attribute_selection == "all"
     end
 
-    def refresh_dialog_fields_service_dialog(service_dialog, data)
+    def refresh_dialog_fields_service_dialog(data)
       data ||= {}
       dialog_fields = Hash(data["dialog_fields"])
       refresh_fields = data["fields"]
       return action_result(false, "Must specify fields to refresh") if refresh_fields.blank?
 
-      define_service_dialog_fields(service_dialog, dialog_fields)
+      service_dialog = define_service_dialog(dialog_fields, data)
```

Comment:
with the service_dialog no longer passed in as parameter (which was RBAC checked earlier for the user via resource_search), is it possible we'll be updating a different service_dialog (data driven and not URL) ?

Reply:
I thought about that too, but the way the classic-ui was doing the refresh before was pulling the dialog off of the resource_action, since it belongs to it. So I guess I could envision a scenario where someone could call, say, `/api/service_dialogs/123` with action set to refresh, and pass in a bogus `resource_action_id` that doesn't correspond to the dialog with the id of 123. We could compare ids, and return an error if they don't match, maybe?

Reply:
maybe pass the service_dialog to refresh_dialog_fields_service_dialog and have it check there against the one defined in data and raise an error accordingly, (with added test for that).  Thanks.